src/audio/aria/aria_hifi3.c

```diff
@@ -102,9 +105,10 @@ void aria_algo_get_data_odd_channel(struct processing_module *mod,
 			/*process data one by one if ch_n is odd*/
 			for (ch = 0; ch < ch_n; ch++) {
 				AE_L32_XP(in_sample, (ae_int32 *)in, inc);
-				out_sample = AE_MULFP32X2RS(in_sample, gain);
-
-				out_sample = AE_SLAA32S(out_sample, att);
+				out1 = AE_MUL32_HH(in_sample, gain);
+				out1 = AE_SRAA64(out1, shift_bits);
+				out1 = AE_SLAA64(out1, 16);
```

Comment:
you can save one shift here, 
1. right shift: 32 - atten 
2. left shift: 16

means you can simply >> 32 - atten - 16, I don't think attenuation > 16, right?

Reply:
The concept of the fix has changed and this problem no longer exists.